### ISSUE TITLE
Wrong brace-style closing brace placement with multiline expressions

### ISSUE THREAD

**Issue by Alaneor** [eslint/eslint#7493]
**Tell us about your environment**

* **ESLint Version:** 3.9.0 (also tested on 3.8.0 with same results)
* **Node Version:** 7.0.0
* **npm Version:** 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```js
'use strict'

module.exports = {
  rules: {
    curly: 1,
    'brace-style': [1, '1tbs'],
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
if (setTimeout) {
  setTimeout(() => true, 1)
  .unref() }
```

**What did you expect to happen?**

I would have expected the closing brace to cause a violation of the `brace-style` rule - closing brace should be placed on the next line after the block body. This is the actual behaviour if I place the `.unref()` together with the `setTimeout()` onto the same line:

```js
if (setTimeout) {
  // Warning: Closing curly brace should be on the same line as opening curly brace
  // or on the line after the previous block
  setTimeout(() => true, 1).unref() }
```

I believe this to be a bug.

**What actually happened? Please include the actual, raw output from ESLint.**

No warning emitted for malformed code.

**Comment by not-an-aardvark** [comment]
Thanks for the report. I was able to reproduce this issue.
